### PR TITLE
add some change scene command 

### PR DESCRIPTION
**Pull request by ohsyme**

### CODE REVIEW

**Comment by gawgua** [comment]
why the second commit delete almost everything

File: src/commands/scene.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    const posData = Interface.target.player.db.posData;
+
+    const _lineup = Interface.target.player.db.lineup;
+    const lineup = _lineup.lineups[_lineup.curIndex];
+    const curAvatarEntity = new ActorEntity(Interface.target.player.scene, lineup.avatarList[0], posData.pos);
+
+    if (!floorID) return c.log("Usage: /scene <floorID>");
+
+    const db = Database.getInstance();
+    await db.update('players',{_id:uid},{posData: { floorID: Number(floorID), planeID: Number(floorID.slice(0,5))}})
```
Comment (memetrollsXD) [inline src/commands/scene.ts:25]: Change in session.player.db.posData and then call player.save()

File: src/commands/scene.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        c.log("No target specified");
+        return;
+    }
+
+    const floorID = (command.args[0]).toString();
+    const uid = Interface.target.player.db._id;
+    const posData = Interface.target.player.db.posData;
+
+    const _lineup = Interface.target.player.db.lineup;
+    const lineup = _lineup.lineups[_lineup.curIndex];
```
Comment (memetrollsXD) [inline src/commands/scene.ts:19]: player.getLineup()

File: src/commands/scene.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import { GetCurSceneInfoScRsp } from "../data/proto/StarRail";
+const c = new Logger("/scene", "blue");
+
+export default async function handle(command: Command) {
+    if (!Interface.target) {
+        c.log("No target specified");
+        return;
+    }
+
+    const floorID = (command.args[0]).toString();
```
Comment (memetrollsXD) [inline src/commands/scene.ts:14]: Why add a toString here? It may be better to just convert it to number here and pass it to the response normally

**Review by memetrollsXD — CHANGES_REQUESTED** [review]

File: src/commands/scene.ts (hunk 3 above)
Comment (gawgua) [reply src/commands/scene.ts:14]: `floorID.slice(0,5)`
number dont support slice()
also it is already a string so i dont see the purpose of toString here

Comment (memetrollsXD) [reply src/commands/scene.ts:14]: We have a MazePlaneExcel util for a reason..

File: src/commands/scene.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        c.log("No target specified");
+        return;
+    }
+
+    const planeID = MazePlaneExcel.fromPlaneId(parseInt(command.args[0]))
+    const uid = Interface.target.player.db._id;
+    const posData = Interface.target.player.db.posData;
+    
+    const lineup2 = await Interface.target.player.getLineup();
+    const curAvatarEntity = new ActorEntity(Interface.target.player.scene, lineup2.avatarList[0].id, posData.pos);
```
Comment (memetrollsXD) [inline src/commands/scene.ts:21]: use leaderSlot

File: src/commands/scene.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+    if (!planeID) return c.log("Usage: /scene <planeID>");
+
+    const db = Database.getInstance();
+    await db.update('players',{_id:uid},{posData: { floorID: Number(planeID!.StartFloorID), planeID: Number(planeID!.PlaneID)}})
```
Comment (memetrollsXD) [inline src/commands/scene.ts:26]: Unnecessary non-null assertion operator + number wrapping

File: src/commands/scene.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    await Interface.target.player.save()
+
+    //ty for tamilpp25 scene
+    Interface.target.send(GetCurSceneInfoScRsp, {
+        retcode: 0,
+        scene: {
+            planeId: Number(planeID!.PlaneID),
+            floorId: Number(planeID!.StartFloorID),
```
Comment (memetrollsXD) [inline src/commands/scene.ts:34]: Unnecessary non-null assertion operator

File: src/commands/scene.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+            entityList: [
+                curAvatarEntity
+            ],
+            entityBuffList: [],
+            entryId: 10001,
+            envBuffList: [],
+            gameModeType: 1,
+            lightenSectionList: []
```
Comment (memetrollsXD) [inline src/commands/scene.ts:42]: Bruteforce some ids here so map lights up

**Review by memetrollsXD — APPROVED** [review]
It works I guess.. here are other things that could be improve for next contributions

**Comment by memetrollsXD** [comment]
You can just resolve the conversations to dismiss and merge the PR

File: src/commands/scene.ts (hunk 4 above)
Comment (tamilpp25) [reply src/commands/scene.ts:21]: its a literal copypasta i did from GetCurSceneInfoCsReq.ts
I will look into improving it

**Comment by ohsyme** [comment]
> dismiss

alr thanks  👍

**Pull request merged**